lib/Parser/Parse.cpp

```diff
@@ -8607,7 +8606,9 @@ ParseNodePtr Parser::ParseExpr(int oplMin,
             *pfLikelyPattern = !!fLikelyPattern;
         }
 
-        if (m_token.tk == tkDArrow)
+        if (m_token.tk == tkDArrow
+            // If we have introduced shorthand error above in the ParseExpr, we need to reset if next token is the assignment.
+            || m_token.tk == tkAsg && oplMin <= koplAsg)
```

Comment:
Clang wants explicit parens around the `m_token.tk == tkAsg && oplMin <= koplAsg`

Reply:
will add it, thanks for look out for those errors